flocker/ca/_ca.py

```diff
@@ -231,14 +231,81 @@ def write_credential_files(self, key_filename, certificate_filename):
             os.umask(original_umask)
 
 
-class NodeCredential(FlockerCredential):
+class UserCredential(PRecord):
+    """
+    A certificate for an API user, signed by a supplied certificate
+    authority.
+
+    :ivar FlockerCredential credential: The certificate and key pair
+        credential object.
+    :ivar bytes username: A username.
+    """
+    credential = field(mandatory=True)
```

Comment:
:question: 

It seems like this field just encourages application code to reach way down into the internals to grab random bits.  This isn't great.  If this were a library we were building for serious long term use, I'd want to address this somehow - probably starting by making this attribute private and carefully scrutinizing any code that wants to use it or attributes of the object it refers to.